# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: mfbt/Array.h
# Commit: 71c122ac0ca7
# Full Hash: 71c122ac0ca73391866b1ef19f4f82bc2d28568b
# Author: David Parks <davidp99@gmail.com>
# Date: 2020-01-09 03:40:18
# Regressor Bug: 1477756
# File Overlap Count: 1
# Description:
#   Bug 1477756 - Initial out-of-process WebGL implementation. r=mccr8,handyman
#   
#   Splits WebGLContext into ClientWebGLContext and HostWebGLContext.  The Client enables the JS-control of a WebGL context in a content procecss while the Host executes the WebGL graphics operations (via a WebGLContext that maintains much of the existing code) in the compositor process.  At this point, the cross-process behavior is disabled -- this series of patches is an incremental step toward that final goal.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D54018
# ==============================================================================

diff -r 5cbf5bb11c58 -r 71c122ac0ca7 mfbt/Array.h
--- a/mfbt/Array.h	Wed Jan 08 21:52:03 2020 +0000
+++ b/mfbt/Array.h	Wed Jan 08 22:19:14 2020 +0000
@@ -19,11 +19,14 @@
 
 namespace mozilla {
 
-template <typename T, size_t Length>
+template <typename T, size_t _Length>
 class Array {
-  T mArr[Length];
+  T mArr[_Length];
 
  public:
+  using ElementType = T;
+  static constexpr size_t Length = _Length;
+
   Array() {}
 
   template <typename... Args>